### PR TITLE
[Travis] enabling MPI

### PR DESCRIPTION
**Pull request by philbucher**
I am just playing around seeing what is possible

Also might be useful for the new CI

### CODE REVIEW

**Comment by loumalouomega** [comment]
Funny thing, this is an MPI related PR and I received the notification twice

**Comment by philbucher** [comment]
the are some actual compiler errors in the MPI-Core, but not sure why (compiles fine on my machine)
Might be related to the version of openMPI

Will investigate further

**Comment by philbucher** [comment]
As suspected, the version of OpenMPI is 1.6.5 which is super old (current version is 4.0.1) and seems to give problems. My local version is 2.1.1 and there it works fine
I will try to update to a newer version but have not yet figured out how

**Comment by philbucher** [comment]
I short: changed from ubuntu 16 to 18, because I could not figure out how to force the installation of the newer version of OpenMPI
We can discuss if we want to go this way, but it is probably not a bad idea to have also the MPI-Core in Travis

File: .travis.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@ sudo: false
 dist: trusty
```
Comment (roigcarlo) [inline .travis.yml]: You should remove this one if switching to 18.04

File: scripts/build/travis/configure_travis_trusty.sh [hunk 2]
```diff
@@ -112,6 +112,7 @@ CMAKE_LIBS=(
 # List of applications to compile. Set to ON/OFF
 
 CMAKE_APPLICATION=(
+  -DMPI_NEEDED=ON
```
Comment (roigcarlo) [inline scripts/build/travis/configure_travis_trusty.sh]: That flag should be in `CMAKE_BUILD` section to avoid confusion (Line 135)

**Review by roigcarlo — CHANGES_REQUESTED** [review]
:+1: Good job!

Just a couple of comments.

**Comment by roigcarlo** [comment]
Also change https://github.com/KratosMultiphysics/Kratos/blob/48bcde3849332a39c9975a840b84133114ffaab8/scripts/build/travis/configure_travis_trusty.sh#L155 to Python 3.6 or it will give problems in the future.

And, maybe its a good idea to rename `configure_travis_trusty` to `configure_travis_bionic` to avoid confusion.

File: .travis.yml (hunk 1 above)
Comment (philbucher) [reply .travis.yml]: good catch, I was actually looking for this but oversaw it ...

Do you see any problems with switching to bionic?

File: scripts/build/travis/configure_travis_trusty.sh (hunk 2 above)
Comment (philbucher) [reply scripts/build/travis/configure_travis_trusty.sh]: makes sense

File: .travis.yml (hunk 1 above)
Comment (roigcarlo) [reply .travis.yml]: No that I can see. We were on trusty mainly because at some point xenial was bugged, but there should be no other reason than this.

**Review by roigcarlo — APPROVED** [review]
:+1: All good to me

**Comment by philbucher** [comment]
BTW I took this (how to use mpi in travis) from the precice guys
=> they seems to use trusty, with OpenMPI version 1.6.5
This version is not working for us (gives compiler errors in the MPI-Core). Since it is super old, I would rather go for a newer version of OpenMPI.
I tried forcing the installation of a newer version in trusty, but I didn't manage => https://github.com/KratosMultiphysics/Kratos/pull/5368/commits/b1ae98f576f72f503ac428e897879eee1b0199a2 and corresponding [travis build](https://travis-ci.org/KratosMultiphysics/Kratos/builds/569287023?utm_source=github_status&utm_medium=notification)
Maybe I did the version enforcing wrong, not sure ...?

Either way, another way of getting it to work in trusty would be to compile OpenMPI ourselves, which I think is sth we can consider in the new CI, but not in Travis

=> hence the easiest solution was to upgrade to bionic, where it directly works

**Comment by roigcarlo** [comment]
Yep the problem probably was that, since ubuntu droped support for 14.04 some milinea ago you have to use unoffical ppa's (I was using sourceline: 'ppa:ubuntu-toolchain-r/test' for the compiler) but I don't know if openmpi is there or somewhere else.

I agree that with the new CI we can build our own, is not hard at all, its what I do in the release machine so it can be redistributed.

**Comment by philbucher** [comment]
ah now I get it, trusty is 14 not 16 🤦‍♂ 
Thanks for the explanation

**Comment by philbucher** [comment]
merging now
Will keep a closer look in the next days, please let me know if problems arise

**Pull request merged**